src/unit/test_mutexqueue.c

```diff
@@ -0,0 +1,351 @@
+#include "../mutexqueue.h"
+#include "test_help.h"
+#include <pthread.h>
+#include <time.h>
+
+/* Helper functions */
+static void add(mutexQueue *q, long value) {
+    unsigned long len = mutexQueueLength(q);
+    mutexQueueAdd(q, (void *)value);
+    TEST_EXPECT(mutexQueueLength(q) == len + 1);
+}
+
+static void padd(mutexQueue *q, long value) {
+    unsigned long len = mutexQueueLength(q);
+    mutexQueueAddPriority(q, (void *)value);
+    TEST_EXPECT(mutexQueueLength(q) == len + 1);
+}
+
+static void popTest(mutexQueue *q, long expected) {
+    unsigned long len = mutexQueueLength(q);
+    long value = (long)mutexQueuePop(q, false);
+    TEST_EXPECT(mutexQueueLength(q) == len - 1);
+    TEST_EXPECT(value == expected);
+}
+
+/* Test: simplePushPop */
+int test_mutexQueueSimplePushPop(int argc, char *argv[], int flags) {
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    mutexQueue *q = mutexQueueCreate();
+    TEST_EXPECT(mutexQueueLength(q) == 0ul);
+    add(q, 1);
+    popTest(q, 1);
+    TEST_EXPECT(mutexQueuePop(q, false) == NULL);
+
+    fifo *f = mutexQueuePopAll(q, false);
+    if (f) fifoDelete(f);
+    mutexQueueRelease(q);
+    return 0;
+}
+
+/* Test: doublePushPop */
+int test_mutexQueueDoublePushPop(int argc, char *argv[], int flags) {
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    mutexQueue *q = mutexQueueCreate();
+    add(q, 1);
+    add(q, 2);
+    popTest(q, 1);
+    popTest(q, 2);
+
+    fifo *f = mutexQueuePopAll(q, false);
+    if (f) fifoDelete(f);
+    mutexQueueRelease(q);
+    return 0;
+}
+
+/* Test: priorityOrdering */
+int test_mutexQueuePriorityOrdering(int argc, char *argv[], int flags) {
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    mutexQueue *q = mutexQueueCreate();
+    add(q, 10);
+    padd(q, 1);
+    add(q, 11);
+    padd(q, 2);
+    popTest(q, 1);
+    popTest(q, 2);
+    popTest(q, 10);
+    popTest(q, 11);
+    TEST_EXPECT(mutexQueuePop(q, false) == NULL);
+
+    fifo *f = mutexQueuePopAll(q, false);
+    if (f) fifoDelete(f);
```

Comment:
Remove these 2 lines.  The queue should be empty.
It's verified by `mutexQueueRelease()` which ensures the queue is empty.